### ISSUE TITLE
Talk Proposal: Frontend Application Bundles, the Docker of Frontend

### ISSUE THREAD

**Issue by geelen** [berlinjs/jsconf2019-special#12]
Frontend apps are on the cusp of a revolution—React is about to launch a streaming server renderer that massively improves performance on both low-end and high-end devices. It's a new dawn for performance & accessibility, but raises another question? How should we host the server component? Can we do that without vendor lock-in?
 
Enter FABs (Frontend Application Bundles), a portable, compact, immutable, server-side JS format that lets you deploy anything from static sites or Next.js to full custom server-rendered React apps anywhere that can run JS, like Cloudflare Workers, Lambda@Edge, Now, etc. Finally, we can have a toolchain that supports the full breadth of frontend projects & server environments, without locking everything to a single host.

---

My talk aims to draw parallels between the server world of a decade ago & the frontend world of today. In the past, companies like AWS & Heroku were innovating & offering exciting new products, but users had to make a choice of which one to use and lock-in their development to target that platform. Today, it's exciting offerings from platforms like Now and Netlify, but all their innovation is locked down inside their ecosystem.

Docker changed that in a profound way, and FABs are poised to do the same thing. So this talk is a prediction of the future—that with React's new streaming renderer, a lot more companies will be looking to add a server-side component, and a standard, interoperable bundle format like FABs makes that much more accessible.

The project is still in its early stages, and this talk will be a draft of the proper FAB "launch" at React Rally later this year, but I'd love the chance to put this new idea out there and get feedback from a new audience!

**Comment by carolstran** [comment]
Hey @geelen! Thanks for submitting, we have you down. We'll be looking over the proposals within the next few days and then we'll let you know 🕺 

Quick question for you though: How long would the talk be?

**Comment by geelen** [comment]
As long as you'd like! The full talk will be 30 minutes but I can get it down to 15 no problem. Maybe even 10 if I you were doing a lightning talk style evening...

**Comment by rmehner** [comment]
@geelen Heck, I wanna see what you're coming up with again! Unfortunately we can't offer you the full talk slot (so many great proposals received us, hard to pick), but we have some 10 minute slots that we need to decide on. Will vote on that asap. Since you already said that you would be down for a 10 minute version of the talk, I'll put you on that list as well :)

**Comment by geelen** [comment]
Hey Robin! No problem at all, I've spoken tons, more than happy to let other people take the limelight. But if there's a 10-minute slot let me know, I'll come up with a super quick demo of what I'm working on, with kinda none of the detail but hopefully hit the cool bits!

**Comment by geelen** [comment]
Also if I can't do a talk @rmehner I'll still find some time over the weekend to show you https://fab.dev in-person :)

**Comment by rmehner** [comment]
Vote's over, you made it :) You have 10 minutes to convince your audience ;-)

**Comment by carolstran** [comment]
Hey @geelen, can you confirm that you got this? We've also added you to the website and will announce asap 🎉 https://jsconf.berlinjs.org/

**Comment by geelen** [comment]
Yep, confirmed. I am still trying to figure out timing, gotta be over at the venue to help out both right before _and_ right after the meetup, but that's ok, I'll make it happen!

**Comment by carolstran** [comment]
Yay thanks! So then maybe putting you in the middle of the lineup would be best? We're happy to put you wherever you prefer, just let us know 😊

**Comment by geelen** [comment]
Middle is perfect. I have to be there for a soundcheck at 6pm and then back to help on the registration at 9. Busy evening!
